client/account/security/TwoFactorTOTP.js

```diff
@@ -0,0 +1,198 @@
+import React, { useState, useCallback, useEffect, useRef } from 'react';
+import { Box, Button, TextInput, Icon, ButtonGroup, Margins } from '@rocket.chat/fuselage';
+import qrcode from 'yaqrcode';
+
+import { useToastMessageDispatch } from '../../contexts/ToastMessagesContext';
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useForm } from '../../hooks/useForm';
+import { useMethod } from '../../contexts/ServerContext';
+import { Modal } from '../../components/basic/Modal';
+
+const mapCodes = (codes) => codes.map((code, index) => ((index + 1) % 4 === 0 ? `${ code }\n` : code)).join(' ');
+
+const BackupCodesModal = ({ codes, onClose, ...props }) => {
+	const t = useTranslation();
+
+	return <Modal {...props}>
+		<Modal.Header>
+			<Icon name='info' size={20}/>
+			<Modal.Title>{t('Backup_codes')}</Modal.Title>
+			<Modal.Close onClick={onClose}/>
+		</Modal.Header>
+		<Modal.Content fontScale='p1'>
+			<Box mb='x8' withRichContent dangerouslySetInnerHTML={{ __html: t('Make_sure_you_have_a_copy_of_your_codes', { codes }) }} />
+		</Modal.Content>
+		<Modal.Footer>
+			<ButtonGroup align='end'>
+				<Button primary onClick={onClose}>{t('Ok')}</Button>
+			</ButtonGroup>
+		</Modal.Footer>
+	</Modal>;
+};
+
+const VerifyCodeModal = ({ onVerify, onCancel, ...props }) => {
+	const t = useTranslation();
+
+	const ref = useRef({});
```

Comment:
```suggestion
	const ref = useRef();
```

We should try to keep using the same type on values attributed to ref objects.